### ISSUE TITLE
Poperties with JsonIgnoreCondition.WhenWritingDefault are not included in the contract

### ISSUE THREAD

**Issue by flibustier7seas** [domaindrivendev/Swashbuckle.AspNetCore#2285]
Properties annotated with `JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingDefault)` attribute are not included in the contract.

### csproj

```xml
<Project Sdk="Microsoft.NET.Sdk.Web">

    <PropertyGroup>
        <TargetFramework>netcoreapp3.1</TargetFramework>
        <LangVersion>8</LangVersion>
        <Nullable>enable</Nullable>
    </PropertyGroup>

    <ItemGroup>
        <PackageReference Include="Swashbuckle.AspNetCore" Version="6.2.3"/>
        <PackageReference Include="System.Text.Json" Version="5.0.2"/>
    </ItemGroup>

</Project>
```

### Models

```csharp
public class User
{
    public Guid Id { get; set; }
    
    [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingDefault)]
    public bool IsDeleted { get; set; }
}
```

### Actual schemas

```json
{
  "schemas": {
    "User": {
      "type": "object",
      "properties": {
        "id": {
          "type": "string",
          "format": "uuid"
        }
      },
      "additionalProperties": false
    }
  }
}
```

### Expected schemas

```json
{
  "schemas": {
    "User": {
      "type": "object",
      "properties": {
        "id": {
          "type": "string",
          "format": "uuid"
        },
        "isDeleted": {
          "type": "boolean",
          "nullable": true
        }
      },
      "additionalProperties": false
    }
  }
}
```

**Comment by Confusingboat** [comment]
This is only an issue if your web app is below .NET 5 since framework version is being used as a proxy for which version of `System.Text.Json` you're expected to be on. Here is the relevant code that respects the conditions:

https://github.com/domaindrivendev/Swashbuckle.AspNetCore/blob/8f363f7359cb1cb8fa5de5195ec6d97aefaa16b3/src/Swashbuckle.AspNetCore.SwaggerGen/SchemaGenerator/JsonSerializerDataContractResolver.cs#L150-L163

I understand this issue is quite old, but I don't think any change here is required.

For anyone encountering this today, the fix is to

- Upgrade your web app to a supported version of .NET
  - 6 or higher at the time of this comment
- Make sure you're on a recent version of `Swashbuckle.AspNetCore`
  - looks like 6.1.2 is the minimum, here is the original PR: https://github.com/domaindrivendev/Swashbuckle.AspNetCore/pull/2049

**Comment by martincostello** [comment]
Closing as it looks like this was resolved by #2049. Please open a new issue if the latest version still exhibits this issue.
